lib/exhal/navigation.ex

```diff
@@ -193,4 +197,36 @@ defmodule ExHal.Navigation do
         @client_module.get(client, Link.target_url!(link, tmpl_vars), opts)
     end
   end
+
+  @typep template_vars :: map()
+  @typep poison_options :: map()
+
+  @spec interpret_nav_opts(map()) :: {template_vars(), boolean(), poison_options()}
+  defp interpret_nav_opts(%{} = opts) do
+    {nav_options, poison_options} = Map.split(opts, [:tmpl_vars, :strict])
```

Comment:
nice! i didn't know about `Map.split`.

Reply:
Now we just need something like `Map.pop/3` that takes multiple keys somehow.